client/android/ZUMOAPPNAME/app/src/main/java/com/example/zumoappname/ToDoActivity.java

```diff
@@ -183,55 +223,174 @@ public void run() {
                             }
                         }
                     });
-                } catch (Exception e){
-                    createAndShowDialog(e, "Error");
-
+                } catch (final Exception e) {
+                    createAndShowDialogFromTask(e, "Error");
                 }
-
                 return null;
             }
-        }.execute();
+        };
+
+        runAsyncTask(task);
 
         mTextNewToDo.setText("");
     }
 
     /**
-     * Refresh the list with the items in the Mobile Service Table
+     * Mark an item as completed in the Mobile Service Table
+     *
+     * @param item
+     *            The item to mark
+     */
+    public ToDoItem addItemInTable(ToDoItem item) throws ExecutionException, InterruptedException {
+        ToDoItem entity = mToDoTable.insert(item).get();
+        //sync().get(); // offline sync
```

Comment:
This should not be present. Adding this would make application work only when the device is online. We should sync only when refresh button is pressed.